### PR TITLE
Add machine readable output for trigger

### LINKED ISSUE #1095: Add machine readable output (-o flag) to kn trigger describe
use case: `kn trigger describe NAME -oyaml`

see `kn service describe NAME -oyaml` for ref

### PR DESCRIPTION
**Pull request by arghya88**
Signed-off-by: Arghya Sadhu <arghya88@gmail.com>

## Changes

*Add machine readable output (-o flag) to kn trigger describe

## Reference

Fixes #1095 

/lint

### CODE REVIEW

**Comment by arghya88** [comment]
@navidshaikh any idea why the test for `-o yaml` is failing with error

`describe_test.go:91: assertion failed: error is not nil: missing apiVersion or kind; try GetObjectKind().SetGroupVersionKind() if you know the type`

File: pkg/kn/commands/trigger/describe.go [hunk 1]
```diff
@@ -105,3 +135,7 @@ func writeTrigger(dw printers.PrefixWriter, trigger *v1beta1.Trigger, printDetai
 		}
 	}
 }
+
+func extractURL(trigger *v1beta1.Trigger) string {
+	return trigger.Status.SubscriberURI.String()
+}
```
Comment (navidshaikh) [inline pkg/kn/commands/trigger/describe.go:141]: This is not trigger's URL but the subscriber's, and I dont think we should  expose `-o url` to trigger describe at all.
Discussed this with @kameshsampath the other day on slack, wdyt @rhuss ?

**Review by navidshaikh — CHANGES_REQUESTED** [review]

**Comment by navidshaikh** [comment]
> @navidshaikh any idea why the test for -o yaml is failing with error

@arghya88 : The genericclioptions util we use to enable the machine readable output, requires GVK set on the object.

You need to update group version kind of the object you `GET` via `knEventingClient`.

For example: [here](https://github.com/knative/client/blob/10128901aa704abc376eb1e586636c871e0a37d2/pkg/eventing/v1beta1/client.go#L109) its done for the list object of trigger and each individual items in the list. For `GET` call you should update GVK for the received object.

here is another example of setting GVK for service object we `GET`
https://github.com/knative/client/blob/10128901aa704abc376eb1e586636c871e0a37d2/pkg/serving/v1/client.go#L193

File: pkg/kn/commands/trigger/describe.go (hunk 1 above)
Comment (rhuss) [reply pkg/kn/commands/trigger/describe.go:141]: I agree, that we should only expose `-o url` to "Addressable" e.g. for objects that have a `status.uri`

Comment (arghya88) [reply pkg/kn/commands/trigger/describe.go]: Got it.Thanks

**Review by navidshaikh — CHANGES_REQUESTED** [review]
Please run `./hack/build.sh` and commit

```
📖 Docs ERROR: Modified files found:  M docs/cmd/kn_trigger_describe.md ERROR: Diff diff --git a/docs/cmd/kn_trigger_describe.md b/docs/cmd/kn_trigger_describe.md index 3884b35b..bdc194a7 100644 --- a/docs/cmd/kn_trigger_describe.md +++ b/docs/cmd/kn_trigger_describe.md @@ -19,9 +19,6 @@ kn trigger describe NAME      # Describe a trigger 'my-trigger' in YAML format    kn trigger describe my-trigger -o yaml - -  # Print only trigger URL -  kn trigger describe my-trigger -o url  ```    ### Options ERROR: /home/prow/go/src/knative.dev/client is out of date. Please run ./hack/build.sh -c and commit.
--
```

**Comment by arghya88** [comment]
@navidshaikh the tests still failed with same error even after adding gvk

**Comment by arghya88** [comment]
@navidshaikh tests are passing now.PTAL

File: CHANGELOG.adoc [hunk 2]
```diff
@@ -27,6 +27,11 @@
 | https://github.com/knative/client/pull/1118[#1118]
 |===
 
+| 🎁
+| Add machine readable output (-o flag) to kn trigger describe
+| https://github.com/knative/client/pull/1117[#1121]
```
Comment (navidshaikh) [inline CHANGELOG.adoc:32]:
```suggestion
| https://github.com/knative/client/pull/1121[#1121]
```

**Review by navidshaikh — CHANGES_REQUESTED** [review]
lgtm, lets fix the typo in CHANGELOG and this should be good to go.

File: CHANGELOG.adoc (hunk 2 above)
Comment (arghya88) [reply CHANGELOG.adoc]: Done

**Comment by arghya88** [comment]
@navidshaikh Changes done.PTAL

**Review by navidshaikh — CHANGES_REQUESTED** [review]
@arghya88 : The CI failure needs a fix in eventing nightly setup, see https://github.com/knative/hack/pull/24  I will retest once its merged/fixed. Meanwhile can you please fix the CHANGELOG as suggested below?

File: CHANGELOG.adoc [hunk 3]
```diff
@@ -30,6 +30,11 @@
 | 🎁
 | Add machine readable output (-o flag) to kn broker describe
 | https://github.com/knative/client/pull/1124[#1124]
+===
```
[the remark this answers is not in the document]
Comment (arghya88) [reply CHANGELOG.adoc]: Done

**Review by navidshaikh — APPROVED** [review]
/retest
/lgtm
/approve

**Pull request merged**